### PR TITLE
feat: always pull latest image

### PR DESCRIPTION
**Pull request by tedim52**
## Description:
Prior to this PR, Kurtosis would use the local image if it was available, and if it wasn't, pulled the image from Dockerhub (or other container registry). This PR reverses that behavior by always attempting to pull the latest image, and using the local image, if available, only if the pull fails. For users, this means they will always be using the latest version of the images they're working with. 

Through testing, I've determined the impact on speed is negligible but I've kept a `FetchLocalImage` function with the old behavior in case we need to rollback this change. I've also removed a redundant `FetchImage` from the `StartServices` as it is already being called for each service during validation phase and in `CreateAndStartContainerArgs`.

## Is this change user facing?
NO

## References
https://github.com/kurtosis-tech/kurtosis/issues/1103

### CODE REVIEW

File: container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go [hunk 1]
```diff
@@ -1121,7 +1121,46 @@ func (manager *DockerManager) GetContainersByLabels(ctx context.Context, labels
...
+	// this allows us to be deterministic in our behaviour
+	if !strings.Contains(dockerImage, dockerTagSeparatorChar) {
+		dockerImage = dockerImage + dockerTagSeparatorChar + dockerDefaultTag
+	}
+	logrus.Infof("Checking if image '%v' is available locally...", dockerImage)
+	doesImageExistLocally, err := manager.isImageAvailableLocally(ctx, dockerImage)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred checking for local availability of Docker image '%v'", dockerImage)
+	}
+	logrus.Infof("Is image available locally?: %v", doesImageExistLocally)
```
Comment (adschwartz) [inline container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go]: Is this for debugging?

Comment (tedim52) [reply container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go]: yes I should change that log level to trace

File: container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go [hunk 2]
```diff
@@ -1121,7 +1121,46 @@ func (manager *DockerManager) GetContainersByLabels(ctx context.Context, labels
 	return result, nil
 }
 
+// [FetchImage] always attempts to retrieve the latest [dockerImage].
+// If retrieving the latest [dockerImage] fails, the local image will be used.
+// Returns error, if no local image is available after retrieving latest fails.
 func (manager *DockerManager) FetchImage(ctx context.Context, dockerImage string) error {
```
Comment (adschwartz) [inline container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go:1127]: @tedim52 is there any way to return the result of fetching the image? It seems a little strange to call this function but not knowing if it was fetched successfully or not (other than the absence of the error)? Maybe you are limited by what the `isImageAvailableLocally` and `pullImage` functions return, so maybe there's not much you can do, but maybe worth having a look at?

**Review by adschwartz — APPROVED** [review]

File: container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go (hunk 2 above)
Comment (tedim52) [reply container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go:1127]: great point, send some ideas over slack -> lmk what you think and i'm happy to make this change. gonna merge for now and open another PR for this

**Comment by mieubrisse** [comment]
I added "user-facing" because this does indeed affect users because it's a potentially-significant change in beahviour. @leeederek FSA for the announce!

**Pull request merged**